packages/smooth_app/lib/main.dart

```diff
@@ -253,23 +257,18 @@ class _SmoothAppState extends State<SmoothApp> {
 /// Layer needed because we need to know the language. Language isn't available
 /// in the [context] in top level widget ([SmoothApp])
 class SmoothAppGetLanguage extends StatelessWidget {
-  const SmoothAppGetLanguage(this.appWidget);
+  const SmoothAppGetLanguage(this.appWidget, this.userPreferences);
 
   final Widget appWidget;
+  final UserPreferences userPreferences;
 
   @override
   Widget build(BuildContext context) {
     // TODO(monsieurtanuki): refactor removing the `SmoothAppGetLanguage` layer?
-    // will refresh each time the language changes
-    context.select(
-      (final UserPreferences userPreferences) =>
-          userPreferences.appLanguageCode,
-    );
```

Comment:
We didn't need this watch here as we have one in main.dart 221 also before this layer